### PR TITLE
Automated Changelogs for Nightly CD

### PR DESCRIPTION
**Pull request by akarpz**
### What does this PR do?

Creates a gitlab CI job dependent on the execution of the nightly deployment that posts a changelog of commits that occurred between the current and last nightly deployment. The message is posted in `#system-probe-ops`. The format of the changelog looks like this:
```
Changelog for commit range: 766da29 to b6c8542c
Add support for LPM_TRIE maps to ebpf check (#19688) @Bryce Kahle
...
:wave: Authors, please check relevant dashboards for issues
```
This will tag all of the authors who contributed in the relevant commit range on slack, so they're aware their changes are about to be released.

### Motivation

https://datadoghq.atlassian.net/browse/NPM-2757

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: requirements.txt
Comment (amenasria) [inline requirements.txt]: **🥜 nitpick:** ‏It'd be better to add it in buildimages. Also for `codeowners` it might be already installed in the pipeline at runtime, but you'd have to double check I might be wrong !

File: tasks/pipeline.py
Comment (akarpz) [reply tasks/pipeline.py:473]: thanks! I'll use that instead

File: requirements.txt
Comment (akarpz) [reply requirements.txt]: I did remove this, but yes unfortunately I tried a test pipeline with this on friday and it failed: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/334810858

File: tasks/libs/pipeline_notifications.py
Comment (amenasria) [inline tasks/libs/pipeline_notifications.py:83]: **❓ question:** ‏Why do we need to `rstrip` the file ? Just wondering since it was already working before !

File: tasks/pipeline.py
Comment (amenasria) [inline tasks/pipeline.py:9]: **💬 suggestion:** ‏I'm usually trying to avoid using `subprocess` while writing invoke tasks because it's already supported by invoke and much easier to read / maintain. Instead of this you can use `ctx.run` which is done in the other tasks !

File: tasks/pipeline.py
Comment (amenasria) [inline tasks/pipeline.py]: **💬 suggestion:** ‏Ditto (`sp.check_output` --> `ctx.run`)

File: tasks/pipeline.py
Comment (amenasria) [inline tasks/pipeline.py]: **💬 suggestion:** ‏Ditto (`sp.check_output` --> `ctx.run`)

**Review by amenasria — CHANGES_REQUESTED** [review]
Otherwise LGTM ! I still want to check the 
```
  artifacts:
    untracked: true
```
because I never saw it before in the codebase !

File: .gitlab/internal_kubernetes_deploy.yml
Comment (akarpz) [reply .gitlab/internal_kubernetes_deploy.yml:64]: yes that's a good suggestion

Comment (akarpz) [reply .gitlab/internal_kubernetes_deploy.yml:64]: Unfortunately it looks like inv is not installed on the `slack-notifier` image: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/337484145

File: tasks/libs/pipeline_notifications.py
Comment (akarpz) [reply tasks/libs/pipeline_notifications.py:83]: I will try removing this and see if it works! was trying to be safe

File: tasks/pipeline.py
Comment (akarpz) [reply tasks/pipeline.py:9]: yes will change this

File: tasks/pipeline.py
Comment (akarpz) [inline tasks/pipeline.py:549]: this command will currently fail as permissions are still being worked out for this SSM param, should be fixed shortly

File: tasks/pipeline.py
Comment (brycekahle) [inline tasks/pipeline.py:515]: suggestion: make title the link to the PR
```suggestion
            message = f"<{url}|{title}> - {author}"
```

File: tasks/pipeline.py
Comment (brycekahle) [inline tasks/pipeline.py:538]: It looks like can you pass multiple emails to `email2slackid`

File: tasks/pipeline.py
Comment (akarpz) [reply tasks/pipeline.py:544]: rate limits on the slack API side (I saw these when testing)

Comment (brycekahle) [reply tasks/pipeline.py:544]: Doesn't this hit our own SDM API? https://github.com/DataDog/images/blob/a4ba01676e392a21d06e80c1310e232e0a252bf3/slack-notifier/sdm/email2slackid#L54

File: tasks/pipeline.py
Comment (akarpz) [reply tasks/pipeline.py:538]: rewriting this to take advantage of this will likely speed up runtime considerably, assuming `email2slackid` is hitting the slack ID in bulk in that case

File: .gitlab/internal_kubernetes_deploy.yml
Comment (brycekahle) [inline .gitlab/internal_kubernetes_deploy.yml:8]: I think it would be better to add specific files to artifacts, rather than all untracked files.

Comment (akarpz) [reply .gitlab/internal_kubernetes_deploy.yml:8]: we won't need this once it all runs on the `slack-notifier` image, this was just a solution to pass data between the different images

File: tasks/pipeline.py
Comment (brycekahle) [inline tasks/pipeline.py:522]: Could the commit SHAs be links too?

Comment (brycekahle) [reply tasks/pipeline.py:522]: or the range, like this: https://github.com/DataDog/datadog-agent/compare/d072b59b...62f4a972

Comment (akarpz) [reply tasks/pipeline.py:522]: yeah I think the range is best, although this is filtering for changes in the `system-probe` space

**Review by amenasria — APPROVED** [review]
Left a few comments. Otherwise LGTM if it works in the CI !

File: tasks/pipeline.py
Comment (akarpz) [reply tasks/pipeline.py:521]: good note, but yes that is the purpose for the sleep :)

File: tasks/pipeline.py
Comment (brycekahle) [inline tasks/pipeline.py:520]: Is it worth trying to cache the results of running `email2slackid` in this function, so it only hits the API when necessary?

File: tasks/pipeline.py
Comment (brycekahle) [inline tasks/pipeline.py:526]:
```suggestion
            message_link = f"- <{url}|{title}>"
```

Comment (brycekahle) [reply tasks/pipeline.py:526]: This will let slack do the formatting

File: tasks/pipeline.py
Comment (brycekahle) [inline tasks/pipeline.py:527]: This `if` could go much higher after the `parse` to prevent unnecessary work:
```
if "dependabot" in author_email or "github-actions" in author_email:
    continue
```

File: tasks/pipeline.py
Comment (brycekahle) [inline tasks/pipeline.py:553]: What happens if the sending of slack message fails?

File: tasks/pipeline.py
Comment (brycekahle) [inline tasks/pipeline.py:519]: suggestion: invert with continue to prevent lots indentation:
```
if not is_system_probe(owners, files):
    continue
```

File: tasks/pipeline.py
Comment (akarpz) [reply tasks/pipeline.py:526]: I did try this earlier and it just sent the hyphen verbatim for some reason: https://dd.slack.com/archives/C05MH06TS8N/p1696968462536479

Comment (brycekahle) [reply tasks/pipeline.py:526]: strange...OK

File: tasks/pipeline.py
Comment (akarpz) [reply tasks/pipeline.py:553]: I'll change this at least to try posting to slack before updating state so it can be rerun manually

File: tasks/pipeline.py
Comment (akarpz) [reply tasks/pipeline.py:526]: trying this one more time with your other suggestions 🤷

File: tasks/pipeline.py
Comment (akarpz) [reply tasks/pipeline.py:527]: I think we still want to include the commit here, I was doing this to exclude the author info because it's a bot account that has a really long email

Comment (akarpz) [reply tasks/pipeline.py:527]: I cleaned the logic up slightly though

Comment (brycekahle) [reply tasks/pipeline.py:527]: You probably don't want to run `email2slackid` on it, rightt?

Comment (akarpz) [reply tasks/pipeline.py:527]: ah I see what you mean.. updated

**Pull request merged**
